### ISSUE TITLE
[BUG] [RC1 Sodium] msgpack requirements are incorrectly pinned <1.0.0

### ISSUE THREAD

**Issue by waynew** [saltstack/salt#57536]
**Description**

We've merged https://github.com/saltstack/salt/pull/57122 to support msgpack >=1.0, but our requirements files prevent installing with a greater msgpack.

We need to fix that.

**Comment by waynew** [comment]
@sagetherage we have folks asking for this in Sodium. It should be just changing the requirements file.

**Comment by aplanas** [comment]
https://github.com/saltstack/salt/pull/57539

**Comment by prometheanfire** [comment]
can this be reopened since the patch was reverted?

**Comment by cmcmarrow** [comment]
@aplanas
We don’t see this as a bug because no platform “that we support” requires/uses msgpack 1.0.0 and no part of salt needs the upgrade.
Why do you need msgpack 1.0.0?
msgpack 1.0.0 is be worked on in #57571. We currently have 7 – 8 failing test.

**Comment by aplanas** [comment]
@cmcmarrow I am a bit confused. This bug is not mine.

There are some bits related with this topic that are indeed bugs, let me try to clarify:

* We, on SUSE / openSUSE, use Salt in Tumbleweed derivatives, that since some month already have msgpack 1.0.0 installed. Salt was not working with this version, and we fixed this here: https://github.com/saltstack/salt/pull/57122

* One comment from the previous thread is pointing that msgpack >= 1.0.0 can be required to fix a memory leak (https://github.com/saltstack/salt/pull/57122#issuecomment-629676973), and that this patch do not address the issue of removing the requirements.txt cap

* To address the second issue, I created https://github.com/saltstack/salt/pull/57122, that was wrong because the static files actually require to be executed under the systems that represent (macos, windows, etc), and I am only interested on Linux. As a side effect, there are other dependencies that also require < 1.0.0, that are out of my scope. Luckily this patch was reverted and we have now the one that you created in https://github.com/saltstack/salt/pull/57122#issuecomment-629676973

The TL;DR version can be that for me only the first patch is relevant, but if the comment about the memory leak is accurate, there is a bug that removing the cap will fix.

**Comment by prometheanfire** [comment]
The main reason I think the cap needs to be raised is so that openstack can use the newer msgpack (mem leak being one reason).  msgpack was already held back for the last release and I'd rather remove salt as a dependency that openstack projects can use than keep msgpack back for another cycle.  The reason openstack needs salt to update the dependency is because we want to keep openstack co-installable.
